cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarPrototype.java

```diff
@@ -0,0 +1,113 @@
+package hu.uni.miskolc.iit.webdevelopment.examples.cars.model;
+
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+
+/**
+ * Created by tothzs on 2017.11.24..
+ *
+ * CarPrototype represents an abstract car.
+ * It defines attributes the are usually same for each cars of the same type.
+ */
+public class CarPrototype {
+
+    /**
+     * Producer of the given prototype, such as Suzuki, Opel etc.
+     */
+    private CarProducer producer;
+    /**
+     * Model of the car. It depends on the producer. For example, Swift or Astra.
+     */
+    private String model;
+    /**
+     * Define the body of the car.
+     * Some car models are produced with different bodies.
+     * For example, Renault Megan has Sedan and Coupe versions.
+     */
+    private CarBodyStyle bodyStyle;
+    /**
+     * Year when the given model was introduced to the market.
+     */
+    private int yearOfIntroduction;
+    /**
+     * Any note about this prototype.
+     */
+    private String note;
+
+    /**
+     * Weight of the empty car.
+     * It is defined in kg.
+     */
+    private int crubWeight;
+    /**
+     * Maximum weight of the vehicle in kg.
+     */
+    private int grossVehicleWeight;
+    /**
+     * number of doors.
+     * It may depends on the body type.
+     */
+    private int doorNumber;
+    /**
+     * Performance of the engine in kW.
+     */
+    private double enginePerformance;
+
+
+
+    public CarPrototype(CarProducer producer, String model, CarBodyStyle bodyStyle, int yearOfIntroduction, String note, int crubWeight, int grossVehicleWeight, int doorNumber, double enginePerformance) {
```

Comment:
negative door number for example?

Reply:
data type of door number has been changed to enum